File: m3u_proxy_frontend/src/types/channel.ts
// 频道数据类型
export interface Channel {
  id: string;
  tvg_id: string;
  name: string;
  logo?: string;
  group: string;
  url: string;
  stream_type: 'HLS' | 'MP4' | 'FLV' | 'Other';
}

// 频道列表响应
export interface ChannelsResponse {
  total: number;
  channels: Channel[];
}

// 分组列表响应
export interface GroupsResponse {
  groups: string[];
}

// 播放信息
export interface PlayInfo {
  id: string;
  name: string;
  logo?: string;
  group: string;
  stream_type: string;
  play_url: string;
  original_url: string;
}

// API 查询参数
export interface ChannelQuery {
  group?: string;
  search?: string;
}

File: m3u_proxy/src/models/channel.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Channel {
    pub id: String,
    pub tvg_id: String,
    pub name: String,
    pub logo: Option<String>,
    pub group: String,
    pub url: String,
    pub stream_type: StreamType,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "lowercase")]
pub enum StreamType {
    HLS,
    MP4,
    FLV,
    Other,
}

impl Channel {
    pub fn detect_stream_type(url: &str) -> StreamType {
        let url_lower = url.to_lowercase();
        if url_lower.contains(".m3u8") || url_lower.contains("m3u8") {
            StreamType::HLS
        } else if url_lower.ends_with(".mp4") {
            StreamType::MP4
        } else if url_lower.ends_with(".flv") || url_lower.contains("flv") {
            StreamType::FLV
        } else {
            StreamType::Other
        }
    }
}
